src/handle_provide_parameter.c

```diff
@@ -727,11 +731,46 @@ static void handle_execute(ethPluginProvideParameter_t *msg, context_t *context)
             }
             break;
 
+            // ###################
+            // Parsing PAY_PORTION
+            // ###################
+
+        case INPUT_PAY_PORTION_LENGTH:
+            context->next_param = INPUT_PAY_PORTION_TOKEN;
+            break;
+        case INPUT_PAY_PORTION_TOKEN:
+            context->next_param = INPUT_PAY_PORTION_RECIPIENT;
+            break;
+        case INPUT_PAY_PORTION_RECIPIENT:
+            context->next_param = INPUT_PAY_PORTION_AMOUNT;
+            break;
+        case INPUT_PAY_PORTION_AMOUNT:
+            // empty statement because switch / case is wonky
+            ;
+            uint8_t decoding[sizeof(context->pay_portion_amount)];
+            for (uint8_t i = 0; i < sizeof(decoding); ++i) {
+                decoding[i] = msg->parameter[PARAMETER_LENGTH -1 - i];
+            }
+            memmove(&context->pay_portion_amount, decoding, sizeof(context->pay_portion_amount));
```

Comment:
Why using `decoding` temporary array instead of directly copying in `pay_portion_amount` field?

Reply:
pay_portion_amount is a `uint16_t`
But this code is defunct since I've added the sum